pkg/reconciler/cronjobsource/cronjobsource.go

```diff
@@ -122,51 +123,61 @@ func (r *Reconciler) Reconcile(ctx context.Context, key string) error {
 	return err
 }
 
-func (r *Reconciler) reconcile(ctx context.Context, cronjob *v1alpha1.CronJobSource) error {
+func (r *Reconciler) reconcile(ctx context.Context, source *v1alpha1.CronJobSource) error {
 	// This Source attempts to reconcile three things.
 	// 1. Determine the sink's URI.
 	//     - Nothing to delete.
 	// 2. Create a receive adapter in the form of a Deployment.
 	//     - Will be garbage collected by K8s when this CronJobSource is deleted.
 	// 3. Create the EventType that it can emit.
 	//     - Will be garbage collected by K8s when this CronJobSource is deleted.
-	cronjob.Status.ObservedGeneration = cronjob.Generation
-
-	cronjob.Status.InitializeConditions()
-
-	sinkObjRef := cronjob.Spec.Sink
-	if sinkObjRef.Namespace == "" {
-		sinkObjRef.Namespace = cronjob.Namespace
+	source.Status.ObservedGeneration = source.Generation
+
+	source.Status.InitializeConditions()
+
+	dest := source.Spec.Sink.DeepCopy()
```

Comment:
I recommend something like:

```golang
	dest := pkgv1alpha1.Destination{
		Ref: destination.GetRef(),
		URI: destination.URI,
	}
	if dest.Ref != nil && dest.Ref.Namespace == "" {
		dest.Ref.Namespace = source.Namespace
	}
```

Reply:
What is here is what @grantr did for apiserversource.

Reply:
I'd be fine with either. This code will need to be revisited anyway when Deprecated fields and methods like `GetRef` go away.

Reply:
Can we do it as a followup and do all three?